test/syncDomains.ts

```diff
@@ -0,0 +1,175 @@
+import { ethers } from "ethers";
+import * as dotenv from "dotenv";
+import * as fs from "fs";
+import {
+  Config,
+  configuration,
+  createInstance,
+  Domain,
+  Instance,
+} from "../src";
+import { Registrar, Registrar__factory } from "../src/contracts/types";
+
+dotenv.config();
+
+const sleep = (m: number): Promise<void> =>
+  new Promise((r) => setTimeout(r, m));
+
+const getLabel = (name: string) => {
+  if (name === null) return "";
+  const labels = name.split(".");
+  return labels.length > 0 ? labels[labels.length - 1] : "";
+};
+
+const getAllDomains = async (): Promise<Domain[]> => {
+  const rinkebyConfig: Config = configuration.rinkebyConfiguration(
+    ethers.providers.getDefaultProvider(ethers.providers.getNetwork("rinkeby"))
+  );
+
+  const zNSInstance = createInstance(rinkebyConfig);
+
+  // list all the domains
+  const domains = await zNSInstance.getAllDomains();
+  // console.log(domains);
+
+  const rootDomain = await zNSInstance.getDomainById(
+    "0x0000000000000000000000000000000000000000000000000000000000000000",
+    false
+  );
+  console.log("rootDomain", rootDomain);
+
+  // sort by domain hierachies
+  domains.sort((domainA: Domain, domainB: Domain) => {
+    const labelsA = domainA.name.split(".");
+    const labelsB = domainB.name.split(".");
+    if (labelsA.length !== labelsB.length) {
+      return labelsA.length < labelsB.length ? -1 : 1;
+    }
+    return domainA.name.localeCompare(domainB.name);
+  });
+
+  return domains;
+};
+
+const getDomainByName = async (
+  zNSInstance: Instance,
+  name: string
+): Promise<Domain | null> => {
+  const domains =
+    name.length < 1 || name === null
+      ? [
+          await zNSInstance.getDomainById(
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
```

Comment:
If you want to get the root, you can do `ethers.constants.HashZero` instead of "0x0000000000000000000000000000000000000000"